src/cmd_sym.ml

```diff
@@ -197,50 +197,49 @@ let cmd profiling debug unsafe optimize workers no_stop_at_failure no_values
   let result = List.fold_left (run_file ~unsafe ~optimize) pc files in
   let thread : Thread.t = Thread.create () in
   let results = Choice.run ~workers result thread in
-  let* failing =
-    list_fold_left
-      (fun failing (result, thread) ->
-        let pc = Thread.pc thread in
-        let symbols = thread.symbol_set in
-        let model = get_model ~symbols solver pc in
-        let* result, err =
-          match result with
-          | Symbolic_choice.Multicore.EVal (Ok ()) -> Ok (failing, false)
-          | EAssert assertion ->
-            Ok (`EAssert (assertion, model) :: failing, true)
-          | ETrap tr -> Ok (`ETrap (tr, model) :: failing, true)
-          | EVal (Error e) -> Error e
-        in
-        if not no_values then
-          let testcase =
-            List.sort compare (Encoding.Model.get_bindings model)
-            |> List.map snd
-          in
-          let+ () = write_testcase ~dir:workspace ~err testcase in
-          result
-        else Ok result )
-      [] (List.of_seq results)
-  in
   let print_bug = function
     | `ETrap (tr, model) ->
       Format.pp_std "Trap: %s@\n" (Trap.to_string tr);
       Format.pp_std "Model:@\n  @[<v>%a@]@\n"
         (Encoding.Model.pp ~no_values)
-        model
+        model;
+      Format.pp_flush Stdlib.Format.std_formatter ()
     | `EAssert (assertion, model) ->
       Format.pp_std "Assert failure: %a@\n" Expr.pp assertion;
       Format.pp_std "Model:@\n  @[<v>%a@]@\n"
         (Encoding.Model.pp ~no_values)
-        model
-  in
-  let rec print_and_count_failures count = function
-    | [] -> count
-    | hd :: tl ->
-      print_bug hd;
-      let count = succ count in
-      if no_stop_at_failure then print_and_count_failures count tl else count
+        model;
+      Format.pp_flush Stdlib.Format.std_formatter ()
+  in
+  let rec print_and_count_failures count results =
+    match results () with
+    | Seq.Nil -> Ok count
+    | Seq.Cons ((result, thread), tl) ->
+      let pc = Thread.pc thread in
+      let symbols = thread.symbol_set in
+      let model = get_model ~symbols solver pc in
+      let* count, err =
+        match result with
+        | Symbolic_choice.Multicore.EAssert assertion ->
+          print_bug (`EAssert (assertion, model));
+          Ok (succ count, true)
+        | ETrap tr ->
+          print_bug (`ETrap (tr, model));
+          Ok (succ count, true)
+        | EVal (Ok ()) -> Ok (count, false)
+        | EVal (Error e) -> Error e
+      in
+      if not no_values then
+        let testcase =
+          List.sort compare (Encoding.Model.get_bindings model) |> List.map snd
```

Comment:
I believe `Encoding` is already sorting the elements given by `get_bindings` (I wrote the code to do it IIRC).

Reply:
This `List.sort` was present before this PR.
Has `Encoding` evolved in the meantime? If so, should this part be removed? (See next comment ...)

Reply:
Oh indeed. You can try to remove it and see if it changes anything.

Reply:
Ok, the `List.sort` can be removed. Also the `write_testcase` call just after ? (I don't know what it's for)
I've just tried deleting everything: it's all ok!

Reply:
When this was done encoding wouldn't sort elements. So it can be removed. However, this code is still wrong. 

The outputed testcase  is given is in a lexicographical order, making stuff like `symbol_10` appearing before `symbol_2` which is obviously wrong.

Reply:
But we can use `symbols` to return the correct testcase, something like:

```ocaml
          let testcase =
            List.map
              (fun sym ->
                match Encoding.Model.evaluate model sym with
                | None -> assert false (* unreachable due to line 205 *)
                | Some v -> v )
              symbols
            |> List.rev
          in

```

Reply:
Oh I didn't see the comment about `write_testcase`. We probably don't want to delete it entirely as these are the concrete inputs for the symbolic program. But we should put it under some flag so that they are only generated when its on?

Reply:
I'm not sure I've followed what you guys wanted around the write_testcase code but given that this was here before this PR and is pretty orthogonal I suggest that we make an issue for it and fix it later

Reply:
It was changed to `let _ = write_testcase ...` and I was saying it should stay as before: `let+ () = write_testcase ...` (which is the case now).